weblab_cg/templates/chaste/normal_model.cpp

```diff
@@ -0,0 +1,148 @@
+    }
+    
+    {{class_name}}FromCellML::~{{class_name}}FromCellML()
+    {
+    }
+    
+    double {{class_name}}FromCellML::GetIIonic(const std::vector<double>* pStateVariables)
+    {
+        // For state variable interpolation (SVI) we read in interpolated state variables,
+        // otherwise for ionic current interpolation (ICI) we use the state variables of this model (node).
+        if (!pStateVariables) pStateVariables = &rGetStateVariables();
+        const std::vector<double>& rY = *pStateVariables;
+		{% for state_var in state_vars %}
+		{%- if state_var.in_ionic %}double {{ state_var.var }} = {% if loop.index0 == membrane_voltage_index %}(mSetVoltageDerivativeToZero ? this->mFixedVoltage : rY[{{loop.index0}}]);{%- else %}rY[{{loop.index0}}];{%- endif %}
+		// Units: {{state_var.units}}; Initial value: {{state_var.initial_value}}
+		{% endif %}{%- endfor %}{% for ionic_var in ionic_vars %}
+        const double {{ionic_var.lhs}} = {{ionic_var.rhs}}; // {{ionic_var.units}}
+		{%- endfor %}
+        
+		const double i_ionic = var_chaste_interface__i_ionic;
+        EXCEPT_IF_NOT(!std::isnan(i_ionic));
+        return i_ionic;
+    }
+    
+    void {{class_name}}FromCellML::EvaluateYDerivatives(double var_chaste_interface__environment__time, const std::vector<double>& rY, std::vector<double>& rDY)
+    {
+        // Inputs:
+        // Time units: millisecond
+		{%- for state_var in state_vars %}
+		{% if state_var.in_y_deriv %}double {{ state_var.var }} = {% if loop.index0 == membrane_voltage_index %}(mSetVoltageDerivativeToZero ? this->mFixedVoltage : rY[{{loop.index0}}]);{%- else %}rY[{{loop.index0}}];{%- endif %}{%- endif %}
+		// Units: {{state_var.units}}; Initial value: {{state_var.initial_value}}
+		{%- endfor %}
+        
+        // Mathematics
+		{% for deriv in y_derivative_equations %}{%- if deriv.is_voltage%}double {{deriv.lhs}};{%- endif %}{%- endfor %}
+        {%- for deriv in y_derivative_equations %}{%- if not deriv.in_membrane_voltage %}
+		const double {{deriv.lhs}} = {{deriv.rhs}}; // {{deriv.units}}{%- endif %}
+		{%- endfor %}
+		
+        if (mSetVoltageDerivativeToZero)
+        {
+            {% for deriv in y_derivative_equations %}{%- if deriv.is_voltage%}{{deriv.lhs}} = 0.0;{%- endif %}{%- endfor %}
+        }
+        else
+        {
+			{%- for deriv in y_derivative_equations %}{% if deriv.in_membrane_voltage %}
+			{% if not deriv.is_voltage%}const double {% endif %}{{deriv.lhs}} = {{deriv.rhs}}; // {{deriv.units}}{%- endif %}
+			{%- endfor %}
+        }
+		{% for deriv in y_derivatives %}
+        rDY[{{loop.index0}}] = {{deriv}};
+		{%- endfor %}
+    }
+    
+template<>
+void OdeSystemInformation<{{class_name}}FromCellML>::Initialise(void)
+{
+    this->mSystemName = "{{free_variable.system_name}}";
+    this->mFreeVariableName = "{{free_variable.name}}";
+    this->mFreeVariableUnits = "{{free_variable.units}}";
+    
+    {% for ode_info in ode_system_information %}// rY[{{loop.index0}}]:
```

Comment:
Leftover debug code?

Reply:
entierly legit chaste code (it's the template remember?)

Reply:
Oooooh I see it now!